autonomy/analyse/dialogues.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Analyse dialogue definitions."""
+
+
+import importlib
+import importlib.machinery
+import importlib.util
+import types
+from importlib.machinery import ModuleSpec
+from pathlib import Path
+from typing import Any, Dict, List, cast
+
+import yaml
+
+from autonomy.analyse.constants import CLASS_NAME, DIALOGUES, DIALOGUES_FILE, MODELS
+
+
+DIALOGUES_LOWER = DIALOGUES.lower()
+
+
+def load_dialogues_module_from_skill_path(skill_path: Path) -> types.ModuleType:
+    """Load `dialogues.py` module for the given skill."""
+
+    loader = importlib.machinery.SourceFileLoader(
+        DIALOGUES, str(skill_path / DIALOGUES_FILE)
+    )
+    spec = cast(ModuleSpec, importlib.util.spec_from_loader(DIALOGUES, loader))
+    module = importlib.util.module_from_spec(spec)
+    loader.exec_module(module)
+
+    return module
+
+
+def validate_and_get_dialogues(configuration: Dict[str, Any]) -> Dict[str, str]:
+    """Returns dialogue names to class name mappings"""
+
+    mapping = {}
+    for name, info in cast(Dict[str, Dict[str, str]], configuration[MODELS]).items():
```

Comment:
No need to pass the full config.

```suggestion
def validate_and_get_dialogues(models_configuration: Dict[str, Dict[str, str]]) -> Dict[str, str]:
    """Returns dialogue names to class name mappings"""

    mapping = {}
    for name, info in models_configuration.items():
```